src/shell/initrc.lua

```diff
@@ -0,0 +1,17 @@
+
+-- Load all *.so plugins from plugin.serarchpath
+plugin.load { file = "*.so", conf = {} }
+
+-- Copy jobspec attributes.system.shell.options to shell.options
+jobspec = shell.info.jobspec
+if jobspec.attributes.system.shell then
+    shell.options = jobspec.attributes.system.shell.options or {}
+end
+
+-- Source all rc files under shell.rcpath/lua.d/*.lua of shell rcpath:
+source (shell.rcpath .. "/lua.d/*.lua")
+
+-- Now source user initrc.lua if it exists
+source_if_exists "~/.flux/shell/initrc.lua"
```

Comment:
` ~/.flux/shell/initrc.lua`  is a cool feature!    Maybe this is a more general "user-configurability vs provenance" question, but should we be logging the rc files that a job sources, or maybe the list of plugins that were loaded so that the hotline or whoever has something to go on if the user's rc file or custom plugin stack are causing jobs to fail?

Reply:
Yeah, the logging is another piece of this PR that isn't in place yet.

Provenance is another reason I was thinking it would be nice to add the user initrc to the jobspec, though of course this wouldn't be feasible for custom `.so` plugins, or lua fragments sourced via `source "/path/to/*.lua"`, so there would be some loss of flexibility there. 

At the very least logging to the job shell log should happen, I'm not sure about other provenance ideas -- it could use some discussion.